src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1139,9 +1154,22 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
         if (ASR::is_a<ASR::Function_t>(*s)) {
             ASR::Function_t *func = ASR::down_cast<ASR::Function_t>(s);
             if( n_kwargs > 0 && !is_generic_procedure ) {
-                args.reserve(al, n_pos_args + n_kwargs);
+                args.reserve(al, func->n_args);
                 visit_expr_list(pos_args, n_pos_args, kwargs, n_kwargs,
                                 args, rt_subs, func, loc);
+            } else if (args.size() < func->n_args) {
```

Comment:
After going through the rest of the changes, it seems that this `else if` block might not be required. @assem2002 Can you try removing this `else-if` block and see if things still work?

Reply:
Second thought: It might be needed. It looks like a better place for this can be inside the function `void visit_expr_list(AST::expr_t** exprs, size_t n, Vec<ASR::call_arg_t>& call_args_vec)`, but we can do that later. For now the current place is fine for me.